core/src/main/java/org/ehcache/events/CacheEventNotificationServiceImpl.java

```diff
@@ -122,112 +174,162 @@ public void setStoreListenerSource(Cache<K, V> source) {
   }
 
   @Override
-  public void onEvent(final CacheEvent<K, V> event) {
+  public void onEvent(CacheEvent<K, V> event) {
```

Comment:
would it be acceptable to directly mark the event `FIRED` if no listeners are found at this point?